core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -871,6 +871,7 @@ public PlanOptimizers(
                 ImmutableSet.<Rule<?>>builder()
                         .addAll(simplifyOptimizerRules) // Should be always run after PredicatePushDown
                         .add(new PushPredicateIntoTableScan(plannerContext, typeAnalyzer))
+                        .add(new RemoveRedundantTableScanPredicate(plannerContext, typeAnalyzer))
```

Comment:
This is the only change in "Fix redundant predicate above table scan not being removed" commit.
i think commit boundaries were blurred during cherry of #10549